databases/week2/Part4.sql

```diff
@@ -0,0 +1,133 @@
+-- MySQL Script generated by MySQL Workbench
+-- Fri 16 Oct 2020 01:07:37 PM CEST
+-- Model: New Model    Version: 1.0
+-- MySQL Workbench Forward Engineering
+
+SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0;
+SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0;
+SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION';
+
+-- -----------------------------------------------------
+-- Schema SupplyChain
+-- -----------------------------------------------------
+
+-- -----------------------------------------------------
+-- Schema SupplyChain
+-- -----------------------------------------------------
+CREATE SCHEMA IF NOT EXISTS `SupplyChain` DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci ;
+USE `SupplyChain` ;
+
+-- -----------------------------------------------------
+-- Table `SupplyChain`.`Store`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `SupplyChain`.`Store` (
+  `id` INT NOT NULL,
```

Comment:
it's typical to use auto-increment and unique for the id field

Reply:
@arya56 the reason to use auto incement on the primary key is so the database selects the ID to use when inserting a new row into a table. When not using auto-increment you would have to figure out your own IDs when inserting data into the table.

Imagine 100 users all trying to insert something into the database at the same time, how can you make sure that they are choosing unique IDs if they have to figure them out by themselves? Auto-increment solves this problem :)